src/VisualStudio/Core/Test/Completion/VisualBasicMockCompletionProvider.vb

```diff
@@ -0,0 +1,20 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Completion
+Imports Microsoft.CodeAnalysis.Host.Mef
+
+Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Completion
+    <ExportCompletionProvider(NameOf(VisualBasicMockCompletionProvider), LanguageNames.VisualBasic)>
+    <[Shared]>
+    <PartNotDiscoverable>
```

Comment:
what's the general intuition around a non-discoverable part?  you can explicitly make a mef container that has it in it?  but if you ask for a mef container for an assembly, it won't be found/used automatically?

Reply:
> you can explicitly make a mef container that has it in it? but if you ask for a mef container for an assembly, it won't be found/used automatically?

Exactly 👍